src/Giraffe/TokenParsers.fs

```diff
@@ -92,6 +92,11 @@ let private floatParse (path:string) ipos fpos =
     | '+' -> go (ipos + 1)
     | _ -> go (ipos)
 
+let private guidParse (path:string) ipos fpos = 
```

Comment:
The performance parser is not supposed to allocate (substring), working solution but I have already built a rolling parser 2x-3x faster then below so we can use that if you don't mind?